src/NuGet.Clients/PackageManagement.UI/PackageInstallationInfo.cs

```diff
@@ -79,6 +80,55 @@ public PackageInstallationInfo(NuGetProject project, NuGetVersion version, bool
             UpdateDisplayText();
         }
 
+        private bool _providersLoaderStarted;
+
+        private OtherPackageManagerProviders _providers;
+        public OtherPackageManagerProviders Providers
+        {
+            get
+            {
+                if (!_providersLoaderStarted && ProvidersLoader != null)
+                {
+                    _providersLoaderStarted = true;
+                    Task.Run(async () =>
```

Comment:
why are we starting a task for every package item? I would instead provide a lazy object from the outside (same one for al view models) instead of a list of providers. And encapsulate all the background magic in it.

Now given that it will be the same object, you won't be spinning off tasks on different threads for every item